test/spdy/parser-test.js

```diff
@@ -0,0 +1,217 @@
+var assert = require('assert');
+
+var transport = require('../../');
+var spdy = transport.protocol.spdy;
+
+describe('SPDY Parser', function() {
+  var parser;
+
+  beforeEach(function() {
+    var pool = spdy.compressionPool.create();
+    parser = spdy.parser.create({});
+    var comp = pool.get('3.1');
+    parser.setCompression(comp);
+  });
+
+  function pass(data, expected, done) {
+    parser.skipPreface();
+    parser.write(new Buffer(data, 'hex'), function (err) {
+      if (err) {
+        console.log('It should not err -> ', err)
+      }
+    });
+
+    parser.once('data', function(frame) {
+      // console.log('THE FRAME \n', frame)
+      // if (frame.data) {
+      //  console.log('hey ', frame.data.toString('hex'))
+      // }
+      assert.deepEqual(frame, expected);
+      assert.equal(parser.buffer.size, 0);
+      done();
+    });
+  }
+
+  function fail(data, code, re, done) {
+    parser.skipPreface();
+    parser.write(new Buffer(data, 'hex'), function(err) {
+      console.log('err - ', err) 
+      assert(err);
+      assert(err instanceof transport.protocol.base.utils.ProtocolError);
+      assert.equal(err.code, spdy.constants.error[code]);
+      assert(re.test(err.message), err.message);
+
+      done();
+    });
+  }
+
+  describe('SYN_STREAM', function() {
+    // [x] pass with http header
+    // [~] pass without http header (apparently it isn't supposed to -
+    // github.com/indutny/spdy-transport/issues/1#issuecomment-116108202
+    // [~] fail on stream ID 0
+    // [X] pass on FIN flag
+    // [ ] pass on UNIDIRECIONAL flag
+
+    it('should parse frame with http header', function(done) {
+      var hexFrame =  '800300010000002c0000000100000000000078' +
+                      'f9e3c6a7c202e50e507ab442a45a77d7105006' +
+                      'b32a4804974d8cfa00000000ffff';
+
+      pass(hexFrame, {
+        fin: false,
+        headers: {
+          ':method': 'GET',
+          ':path': '/'
+        },
+        id: 1,
+        path: '/',
+        priority: {
+          exclusive: false,
+          parent: 0,
+          weight: 1.
+        },
+        type: 'HEADERS' // by spec 'SYN_STREAM'
+      }, done);
+    })
+
+    /*
+    it('should fail on stream ID 0', function(done) {
```

Comment:
the parser doesn't fail with stream Id 0. Probably this validation is done in another place, could you confirm @indutny ?